### PR TITLE
Alkwa/cte storybook docs

### PR DESCRIPTION
**Pull request by alkwa-msft**
# What
Adding storybook docs for how to use CTE
I still have some open questions:
- What is the official name for CTE?
- What is the  version of UI Library where it will be available?
- Can we get verification on the snippet used to see if they are correct?

# Why
We didn't have any documentation on how to use this. By adding this content into storybook, developers will understand how to best use CTE with the UI Library

# How Tested
Ran locally.

### CODE REVIEW

**Review by prabhjot-msft — APPROVED** [review]
Looks good to me!

File: packages/storybook/stories/CustomTeamsEndpoint.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import { DetailedBetaBanner } from './BetaBanners/DetailedBetaBanner';
+import CustomTeamsEndpointSnippetText from '!!raw-loader!./snippets/CustomTeamsEndpoint.snippet.tsx';
+
+<Meta
+  id="customTeamsEndpoint"
+  title="Concepts/Custom Teams Endpoint"
+  parameters={{ previewTabs: { canvas: { disable: true, hidden: true } } }}
+/>
+
+# Custom Teams Endpoint
```
Comment (alkwa-msft) [inline packages/storybook/stories/CustomTeamsEndpoint.stories.mdx:11]: Should be Communications as a Teams user

File: packages/storybook/stories/snippets/CustomTeamsEndpoint.snippet.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    } catch {
+      console.error('Failed to construct token credential');
+      return undefined;
+    }
+  }, [props.token]);
+
+  const adapter = useAzureCommunicationCallAdapter(
+    {
+      userId: props.userId,
+      displayName: props.displayName, // Max 256 Characters
```
Comment (PorterNan) [inline packages/storybook/stories/snippets/CustomTeamsEndpoint.snippet.tsx:38]: No displayname needed for teams call adapter

File: packages/storybook/stories/CustomTeamsEndpoint.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+Note: The UI Library is not able to create teams access tokens or interact with your existing tenant.
+
+## How to incorporate Custom Teams Endpoint with the UI Library in your experience.
+
+The following is a code snippet of how to create a custom teams endpoint experience with the Call Composite.
+
+<Source code={CustomTeamsEndpointSnippetText} />
+
+## FAQs
+
```
Comment (PorterNan) [inline packages/storybook/stories/CustomTeamsEndpoint.stories.mdx:42]: Would be nice to point developers to the guide of generating a teams token in FAQ
https://learn.microsoft.com/en-us/azure/communication-services/quickstarts/manage-teams-identity?pivots=programming-language-javascript

**Pull request merged**